test/unittests/commands/xmlupload/test_ontology_client.py

```diff
@@ -0,0 +1,285 @@
+from dataclasses import dataclass
+from test.unittests.commands.xmlupload.connection_mock import ConnectionMockBase
+from typing import Any
+
+from pytest_unordered import unordered
+
+from dsp_tools.commands.xmlupload.ontology_client import (
+    OntologyClientLive,
+    _get_all_classes_from_graph,
+    _get_all_properties_from_graph,
+    _remove_prefixes,
+    format_ontology,
+)
+
+# pylint: disable=missing-class-docstring,missing-function-docstring,unused-argument,redefined-outer-name,protected-access
+
+
+@dataclass
+class ConnectionMock(ConnectionMockBase):
+    get_responses: dict[Any, Any]
+
+    def get(self, route: str, headers: dict[str, str] | None = None) -> dict[Any, Any]:
+        return self.get_responses
+
+
+def test_get_all_classes_from_graph() -> None:
+    test_graph = [
+        {
+            "knora-api:isResourceClass": True,
+            "rdfs:label": "Sequenz einer Audio-Ressource",
+            "knora-api:canBeInstantiated": True,
+            "rdfs:subClassOf": [
+                {"@id": "knora-api:Resource"},
+                {"@type": "owl:Restriction", "owl:onProperty": {"@id": "testonto:hasBounds"}, "owl:cardinality": 1},
+                {
+                    "@type": "owl:Restriction",
+                    "owl:onProperty": {"@id": "testonto:sequenceOfAudio"},
+                    "owl:cardinality": 1,
+                },
+            ],
+            "@type": "owl:Class",
+            "@id": "testonto:AudioSequence",
+        },
+    ]
+    res_cls = _get_all_classes_from_graph(test_graph)
+    assert res_cls == ["testonto:AudioSequence"]
```

Comment:
I'm not 100% sure about this myself, but I feel it should not be necessary to provide "fully realistic" json in cases like this - I would personally leave out the properties that do not matter at all because they will just be ignored.

Reply:
Doing so, would drastically shorten this test file, which would be nice :)

Reply:
I thought I would leave all the top level keys in the file. But I can delete the rdfs:subClassOf elements if you think it makes it too long.

Reply:
again, I'm not 100% sure what the best idea is - both realistic data and brevity are good things. But in my last PR I opted for brevity and it felt good. But it's absolutely up to you

Reply:
Thanks for the comment. I will leave the top level keys, because I do like to be able to somewhat see what we actually get in real life, but delete the longer content.